cloud/keystone.go

```diff
@@ -100,6 +107,7 @@ func (client *keystoneV3Client) VerifyToken(token string) (schema.Authorization,
 	// If system token needed reauth and user token is invalid, err wont be propagated neither by return nor Err field,
 	// but response is nil
 	if err != nil || !ok {
+		log.Error("Invalid token, err: %s, result: %#v", err, tokenResult)
```

Comment:
so now even "true errors" about invalid token will be logged as error - isn't it excessive?

Reply:
If token is invalid it means that something went wrong - probably not in gohan, but somewhere else (e.g. UI). We don't expect the user to enter the token manually, therefore I think it's fine to log it as an error. If this will be troublesome it can always be changed.

Reply:
is expired token also invalid? that happens pretty often since expiration time is 1 hour

Reply:
That is a valid point. We can differentiate using `err` and `ok` and either use different log levels or even different log messages for actual errors and reauth.

Reply:
I think that you misinterpreted what this code does. It handles the case when system token (the one that gohan itself uses) expired and user provided invalid token.

Let me describe `gophercloud` behaviour:
1. fetch info about user token - fails (401 unauthorized) because system token is expired
2. renew system token because 401 was received
3. fetch info about user token - fails with 404, but error is not propagated (`tokenResult.Err` is `nil` and `tokenResult.Body` is `nil`)

Scenario when system token is not expired:
1. fetch info about user token - fails with 404, error is propagated (`tokenResult.Err` is not `nil`)

So, both log messages introduced in this PR should have the same level because they are triggered by the same issue (user provided invalid token), which one is printed depends on the current system state. It was suggested before to use `Error` in such case, which I did.

---

Wow, that's a lot of text about one log message ;)